chatsky/pipeline/service/service.py

```diff
@@ -127,66 +94,15 @@ async def _run_handler(self, ctx: Context, pipeline: Pipeline) -> None:
         else:
             raise Exception(f"Too many parameters required for service '{self.name}' handler: {handler_params}!")
 
-    async def _run_as_actor(self, ctx: Context, pipeline: Pipeline) -> None:
-        """
-        Method for running this service if its handler is an `Actor`.
-        Catches runtime exceptions.
-
-        :param ctx: Current dialog context.
-        """
-        try:
-            await pipeline.actor(pipeline, ctx)
-            self._set_state(ctx, ComponentExecutionState.FINISHED)
-        except Exception as exc:
-            self._set_state(ctx, ComponentExecutionState.FAILED)
-            logger.error(f"Actor '{self.name}' execution failed!", exc_info=exc)
-
-    async def _run_as_service(self, ctx: Context, pipeline: Pipeline) -> None:
-        """
-        Method for running this service if its handler is not an Actor.
-        Checks start condition and catches runtime exceptions.
-
-        :param ctx: Current dialog context.
-        :param pipeline: Current pipeline.
-        """
-        try:
-            if self.start_condition(ctx, pipeline):
-                self._set_state(ctx, ComponentExecutionState.RUNNING)
-                await self._run_handler(ctx, pipeline)
-                self._set_state(ctx, ComponentExecutionState.FINISHED)
-            else:
-                self._set_state(ctx, ComponentExecutionState.NOT_RUN)
-        except Exception as exc:
-            self._set_state(ctx, ComponentExecutionState.FAILED)
-            logger.error(f"Service '{self.name}' execution failed!", exc_info=exc)
-
-    async def _run(self, ctx: Context, pipeline: Pipeline) -> None:
-        """
-        Method for handling this service execution.
-        Executes extra handlers before and after execution, launches `_run_as_actor` or `_run_as_service` method.
-
-        :param ctx: (required) Current dialog context.
-        :param pipeline: the current pipeline.
-        """
-        await self.run_extra_handler(ExtraHandlerType.BEFORE, ctx, pipeline)
-
-        if isinstance(self.handler, str) and self.handler == "ACTOR":
-            await self._run_as_actor(ctx, pipeline)
-        else:
-            await self._run_as_service(ctx, pipeline)
-
-        await self.run_extra_handler(ExtraHandlerType.AFTER, ctx, pipeline)
-
     @property
     def info_dict(self) -> dict:
         """
         See `Component.info_dict` property.
         Adds `handler` key to base info dictionary.
         """
         representation = super(Service, self).info_dict
-        if isinstance(self.handler, str) and self.handler == "ACTOR":
-            service_representation = "Instance of Actor"
-        elif callable(self.handler):
+        # Need to carefully remove this
```

Comment:
Could you elaborate?

Reply:
I think I meant that since ACTOR is no longer a valid `Service`, this should be changed. And because there is really only one type of handler now (Callable), the whole "if-else" block could be removed and the f-string inserted into the dict.update(). It's unlikely the handler will be "unknown", right?

Reply:
Got it.
I think `PipelineComponent` should have a property `computed_name` that returns names produced by `rename_component_incrementing`.
So for service it would be
```
if isfunction(self.handler);
  return self.handler.__name__
else:
  return self.handler.__class__.__name__
```
Then this property can be called in `rename_component_incrementing` and here (`info_dict`).